gotpm/cmd/utils.go

```diff
@@ -96,15 +97,53 @@ func dataInput() io.Reader {
 }
 
 // Get the algorithm for public key.
-func getAlgo(keyAlgo string) (tpm2.Algorithm, error) {
+func getAlgo() (tpm2.Algorithm, error) {
 	switch keyAlgo {
 	case "rsa":
 		return tpm2.AlgRSA, nil
 	case "ecc":
 		return tpm2.AlgECC, nil
-	case "":
-		panic("--algo flag not properly setup")
 	default:
 		return tpm2.AlgNull, fmt.Errorf("invalid argument %q for \"--algo\" flag", keyAlgo)
 	}
 }
+
+func getSRKwithAlgo(rwc io.ReadWriter, algo tpm2.Algorithm) (*tpm2tools.Key, error) {
+	switch algo {
+	case tpm2.AlgRSA:
+		return tpm2tools.StorageRootKeyRSA(rwc)
+	case tpm2.AlgECC:
+		return tpm2tools.StorageRootKeyECC(rwc)
+	default:
+		return nil, fmt.Errorf("Cannot create SRK for the given algorithm.")
```

Comment:
Add the invalid algorithm in the error message. i.e. a `%v` in the `Errorf`

Reply:
Okay, but here the algorithm will be printed as an integer which won't add that much information.

Reply:
That’s ok, we can get %v to work properly in a future pr

Reply:
I think %x hex probably makes more sense here, as in tpm2 those constants are represented by hex

Reply:
Good catch! That’s correct